nntrainer/src/neuralnet.cpp

```diff
@@ -395,10 +411,13 @@ int NeuralNetwork::train(std::vector<std::string> values) {
     return ML_ERROR_INVALID_PARAMETER;
   }
 
-  /** Setup data buffer properties */
-  status = data_buffer->setBatchSize(batch_size);
+  status = setTrainConfig(values);
   NN_RETURN_STATUS();
 
+  /** set batch size just before training */
+  setBatchSize(batch_size);
```

Comment:
Is it ok setBatchSize after the initialization?

Reply:
Yes, initialization only sets the layer-specific information, and this not dependent on the batch size (weights, bias, other parameters). This just adds a constraint that the memory for the inputs/outputs must be allocated at the `train/inference/run` methods than at `init` which I think is even better.